src/components/SavedSearchRenderer.js

```diff
@@ -0,0 +1,253 @@
+// @flow
+import React from 'react';
+import PropTypes from 'prop-types';
+import { withRouter } from 'react-router-dom';
+
+import { Dropdown, Glyphicon, MenuItem, Button, Modal, Row, Col } from 'react-bootstrap';
+
+import Configurable from './Configurable';
+import QueryResponse from '../api/QueryResponse';
+import SavedSearch from '../api/SavedSearch';
+
+type SavedSearchRendererProps = {
+  history: PropTypes.object.isRequired,
+  location: PropTypes.object.isRequired,
+};
+
+type SavedSearchRendererDefaultProps = {};
+
+type SavedSearchRendererState = {
+  response?: QueryResponse,
+  error?: string,
+  showSaveSearchModal: boolean,
+  savedSearchTitle: string,
+  savedSearchList: Array<SavedSearch>,
+};
+
+/**
+ * Component to include in the Masthead for entering the query
+ * to use when searching. Must be inside a Searcher component.
+ */
+class SavedSearchRenderer extends React.Component<
+  SavedSearchRendererDefaultProps,
+  SavedSearchRendererProps,
+  SavedSearchRendererState,
+> {
+  static defaultProps: SavedSearchRendererDefaultProps = {};
+
+  static displayName = 'SavedSearchRenderer';
+
+  static contextTypes = {
+    searcher: PropTypes.any,
+  };
+
+  constructor(props: SavedSearchRendererProps) {
```

Comment:
Because state is not based on props, you don't need a constructor. You can set default state below.